src/build-data/detect_version.cpp

```diff
@@ -35,24 +35,28 @@
 
    Wikipedia has a mapping table from Apple Clang version to underlying
    LLVM version: https://en.wikipedia.org/wiki/Xcode
+
+   We omit any mapping logic for versions before XCode 13 since we no longer
+   support those versions. The only reason to detect them is so we can tell that
+   the version of Clang is too old and stop configuration immediately.
```

Comment:
That answers my question re:XCode from above. 😅

Reply:
We may want to split up Clang vs XCode anyway - they are kind of the same, but in a lot of ways it feels like the difference between GCC and Clang is smaller than the difference between Clang and XCode.